xdsl/dialects/omp.py

```diff
@@ -226,6 +253,149 @@ def parse_parameter(cls, parser: AttrParser) -> VariableCaptureKind:
             return cast(VariableCaptureKind, parser.parse_str_enum(cls.enum_type))
 
 
+@irdl_attr_definition
+class DeclareTargetDeviceTypeAttr(
+    EnumAttribute[DeclareTargetDeviceTypeKind], SpacedOpaqueSyntaxAttribute
+):
+    """
+    Implementation of upstream omp.device_type
+    See external [documentation](https://mlir.llvm.org/docs/Dialects/OpenMPDialect/ODS/#declaretargetdevicetypeattr).
+    """
+
+    name = "omp.device_type"
+
+    @classmethod
+    def parse_parameter(cls, parser: AttrParser) -> DeclareTargetDeviceTypeKind:
+        with parser.in_parens():
+            return cast(
+                DeclareTargetDeviceTypeKind, parser.parse_str_enum(cls.enum_type)
+            )
```

Comment:
```suggestion
            return parser.parse_str_enum(DeclareTargetDeviceTypeKind)
```

Reply:
and similar below

Reply:
Nice, that will simplify all the enum attributes a bit. Don't know why I didn't think to do this 😅